sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/_tests_/student-loan-balances-queue.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,130 @@
+    // Set the ESDC response folder to the files mocks folders.
+    process.env.ESDC_RESPONSE_FOLDER = studentLoanBalancesDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(StudentLoanBalancesScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it("Should add monthly loan balance record for the student when the student matches SIN, DOB, and last name.", async () => {
+    // Arrange
+    // Create student.
+    const fakeUser = createFakeUser();
+    fakeUser.lastName = "LASTNAME";
+    const user = await db.user.save(fakeUser);
+    // Student
+    const student = await saveFakeStudent(
+      db.dataSource,
+      { user },
+      {
+        initialValue: { birthDate: "1998-03-24" },
+        sinValidationInitialValue: { sin: "900041310" },
+      },
+    );
+    // Queued job.
+    const mockedJob = mockBullJob<void>();
+    mockDownloadFiles(sftpClientMock, [STUDENT_LOAN_BALANCES_FILENAME]);
+    // Act
+    const result = await processor.processStudentLoanBalancesFiles(
+      mockedJob.job,
+    );
+    // Assert
+    expect(result.length).toBe(1);
+    expect(
+      mockedJob.containLogMessages([
+        `File contains 1 Student Loan balances.`,
+        `Inserted Student Loan balances file ${STUDENT_LOAN_BALANCES_FILENAME}.`,
+      ]),
+    ).toBe(true);
+    expect(result).toContain("Process finalized with success.");
+    // Expect the file was deleted from SFTP.
+    expect(sftpClientMock.delete).toHaveBeenCalled();
+    const studentLoanBalance = await db.studentLoanBalance.find({
+      select: {
+        balanceDate: true,
+        cslBalance: true,
+      },
+      where: {
+        student: { id: student.id },
+      },
+    });
+    // Expect student loan balance contains the student record.
+    expect(studentLoanBalance).toEqual([
+      {
+        balanceDate: "2023-12-31",
+        cslBalance: 148154,
+      },
+    ]);
+  });
+
+  it("Should not add monthly loan balance record when the student is not found.", async () => {
+    // Arrange
+    // Queued job.
+    const mockedJob = mockBullJob<void>();
+    mockDownloadFiles(sftpClientMock, [STUDENT_LOAN_BALANCES_FILENAME]);
+    // Act
+    const result = await processor.processStudentLoanBalancesFiles(
+      mockedJob.job,
+    );
+    // Assert
+    expect(result.length).toBe(3);
+    expect(result).toContain("Process finalized with success.");
+    expect(
+      mockedJob.containLogMessages([
+        `Error processing file ${STUDENT_LOAN_BALANCES_FILENAME}.`,
+      ]),
+    ).toBe(true);
+    // Expect the file was deleted from SFTP.
+    expect(sftpClientMock.delete).toHaveBeenCalled();
+    const studentLoanBalancesCount = await db.studentLoanBalance.count({
+      where: {
+        student: { id: 25 },
```

Comment:
Should it be adjusted as below?
```ts
where: {
        student: { id: student.id },
```

Reply:
this is just a random id of a student who is not present in the system, to validate the negative scenario.

Reply:
You can create the student as the one from the previous tests and not provide the "LASTNAME" for instance. And then search for the student ID that was inserted.
```ts
const student = await saveFakeStudent(
  db.dataSource,
  undefined,
  {
    initialValue: { birthDate: "1998-03-24" },
    sinValidationInitialValue: { sin: "900041310" },
  },
);
```